lib/Normalize_std_ast.ml

```diff
@@ -287,20 +249,23 @@ let make_mapper conf ~ignore_doc_comments ~erase_jane_syntax =
           (Pat.or_ ~loc:loc1 ~attrs:attrs1
              (Pat.or_ ~loc:loc2 ~attrs:attrs2 pat1 pat2)
              pat3 )
-    | Ppat_constraint (pat1, {ptyp_desc= Ptyp_poly ([], _t); _}) ->
+    | Ppat_constraint (pat1, None, _ :: _) when erase_jane_syntax ->
+        m.pat m pat1
+    | Ppat_constraint (pat1, Some {ptyp_desc= Ptyp_poly ([], _t); _}, _) ->
         (* The parser put the same type constraint in two different nodes:
            [let _ : typ = exp] is represented as [let _ : typ = (exp :
            typ)]. *)
         m.pat m pat1
     | _ -> Ast_mapper.default_mapper.pat m pat
   in
   let typ (m : Ast_mapper.mapper) typ =
-    let typ = {typ with ptyp_loc_stack= []} in
     let typ =
       { typ with
-        ptyp_attributes=
-          convert_legacy_jane_street_local_annotations ~segment:Type
-            typ.ptyp_attributes }
+        ptyp_loc_stack= []
+      ; ptyp_attributes=
+          ( typ.ptyp_attributes
+          |> if erase_jane_syntax then map_attributes_no_sort m else Fn.id )
```

Comment:
This seems like it might treat doc comments wrongly, because the decision of whether to strip them is dependent on `erase_jane_syntax`. We probably don't want to strip doc comments here, but `map_attributes_no_sort` appears to.

Reply:
Not sure I agree with any of your points about doc comments. The mapper removes doc comments when `ignore_doc_comments` is true, the same as it always did. The difference here is that if `erase_jane_syntax` is true, it will do that step right now -- otherwise, it will still do it in the `Ast_mapper.default_mapper.typ` call below (since that call will invoke our `attributes` mapper on `ptyp_attributes`). The reason we need a special case is because `Ast_mapper.default_mapper.typ` hides the jane-syntax related attributes away before we can get to them with the `attributes` mapper.

Reply:
We spoke about this. I'm satisfied.